src/combinations_with_replacement.rs

```diff
@@ -105,32 +106,11 @@ where
             };
         }
 
-        // Check if we need to consume more from the iterator
-        // This will run while we increment our first index digit
-        self.pool.get_next();
-
-        // Work out where we need to update our indices
-        let mut increment: Option<(usize, usize)> = None;
-        for (i, indices_int) in self.indices.iter().enumerate().rev() {
-            if *indices_int < self.pool.len() - 1 {
-                increment = Some((i, indices_int + 1));
-                break;
-            }
+        if self.increment_indices() {
```

Comment:
Maybe `if increment_indices {None} else {Some(...)}`?

Reply:
In the end (not just this commit), the `Some` part is used twice and it makes sense to me.
The way I cut the commits may not have been ideal.